autonomy/cli/scaffold_fsm.py

```diff
@@ -753,6 +756,193 @@ def _add_abstract_round_abci_if_not_present(ctx: Context) -> None:
+    """RoundTestsFileGenerator"""
+
+    FILENAME = "tests_" + ROUNDS_FILENAME
+
+    ROUNDS_FILE_HEADER = dedent(
+        """\
+        \"\"\"This package contains the tests for rounds of {FSMName}.\"\"\"
+
+        from typing import Any, FrozenSet, Hashable, Type, cast
+        from unittest import mock
+
+        import pytest
+
+        # TODO: define and import specific payloads explicitly by name 
+        from packages.{author}.skills.{skill_name}.payloads import *
+        from packages.{author}.skills.{skill_name}.rounds import (
+            Event,
+            SynchronizedData,
+            {non_degenerate_rounds}
+        )
+        from packages.valory.skills.abstract_round_abci.base import (
+            AbciAppDB,
+            AbstractRound,
+            BaseTxPayload,
+            ConsensusParams,
+        )
+
+
+        MAX_PARTICIPANTS: int = 4
+
+    """
+    )
+
+    BASE_CLASS = dedent(
+        """\
+        class BaseRoundTestClass:
+            \"\"\"Base test class for Rounds.\"\"\"
+
+            # TODO: replace by skill-specific base classes
+            round_class: Type[AbstractRound]
+            payload_class: Type[BaseTxPayload]
+            round: AbstractRound
+            payload: BaseTxPayload
+            synchronized_data: SynchronizedData
+            consensus_params: ConsensusParams
+            participants: FrozenSet[str]
+
+            def setup(self, **kwargs: Any) -> None:
+                \"\"\"Setup the test method.\"\"\"
+
+                self.participants = frozenset(f"agent_{i}" for i in range(MAX_PARTICIPANTS))
+                data = dict(participants=self.participants, all_participants=self.participants)
+                data.update(kwargs)
+                self.synchronized_data = SynchronizedData(
+                    AbciAppDB(setup_data=AbciAppDB.data_to_lists(data))
+                )
+                self.consensus_params = ConsensusParams(max_participants=MAX_PARTICIPANTS)
+                self.round = self.round_class(
+                    synchronized_data=self.synchronized_data,
+                    consensus_params=self.consensus_params,
+                )
+
+            def deliver_payloads(self, **content: Hashable) -> SynchronizedData:
+                \"\"\"Deliver payloads\"\"\"
+
+                payloads = [self.payload_class(sender=p, **content) for p in self.participants]
+                first_payload, *payloads = payloads
+                self.round.process_payload(first_payload)
+                assert self.round.collection == {first_payload.sender: first_payload}
+                assert self.round.end_block() is None
+                self._test_no_majority_event(self.round)
+                for payload in payloads:
+                    self.round.process_payload(payload)
+                kwargs = dict(path_selection=self.round.most_voted_payload)
+                return cast(SynchronizedData, self.synchronized_data.update(**kwargs))
```

Comment:
I need to improve this. To my understanding every round is either:
- child of `CollectionRound`
- a child of `OnlyKeeperSendsRound`
- degenerate (no tests)

right now I have not covered the `OnlyKeeperSendsRound`

Reply:
No there are more round classes than these; check abstract_round_abci

Reply:
I did I don't see any others